addons/portal/controllers/portal.py

```diff
@@ -261,6 +261,31 @@ def _update_password(self, old, new1, new2):
 
         return {'success': {'password': True}}
 
+    @route('/my/deactivate_account', type='http', auth='user', website=True, methods=['POST'])
+    def deactivate_account(self, validation, password):
+        values = self._prepare_portal_layout_values()
+        values['get_error'] = get_error
+
+        if validation != request.env.user.login:
+            values['errors'] = {
+                'deactivate': _('You should enter "%s" to validate your action.', request.env.user.login),
+            }
+        else:
+            try:
+                request.env['res.users']._check_credentials(password, {'interactive': True})
+                # A user can not self-deactivate
+                request.env.user.with_user(SUPERUSER_ID)._deactivate_portal_user()
```

Comment:
Can't we simply sudo this?

Would be better to keep track of who introduced the request in the create_uid of "res.users.deletion", otherwise everything will be referenced as requested by the admin.